front/src/modules/trainschedule/hooks/useSelectedTrain.tsx

```diff
@@ -0,0 +1,60 @@
+import { useMemo } from 'react';
+
+import dayjs from 'dayjs';
+import { useSelector } from 'react-redux';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { Train } from 'reducers/osrdconf/types';
+import { getSelectedTrainId } from 'reducers/simulationResults/selectors';
+import { Duration } from 'utils/duration';
+import {
+  extractOccurrenceIndexFromOccurrenceId,
+  extractPacedTrainIdFromOccurrenceId,
+  isOccurrenceId,
+  isPacedTrainResponseWithPacedTrainId,
+  isTrainScheduleId,
+} from 'utils/trainId';
+
+const useSelectedTrain = (): Train | undefined => {
+  const trainId = useSelector(getSelectedTrainId);
+
+  const timetableItemId = useMemo(() => {
+    if (!trainId) return undefined;
+    return isOccurrenceId(trainId) ? extractPacedTrainIdFromOccurrenceId(trainId) : trainId;
+  }, [trainId]);
+
+  const { data: timetableItem } = osrdEditoastApi.endpoints.getTimetableItemById.useQuery(
+    {
+      id: timetableItemId!,
+    },
+    { skip: !timetableItemId }
+  );
+
+  return useMemo(() => {
+    if (!trainId || !timetableItem) return undefined;
+    if (isTrainScheduleId(trainId)) {
+      if (isPacedTrainResponseWithPacedTrainId(timetableItem) || trainId !== timetableItem.id)
+        return undefined;
+      return timetableItem;
+    }
+
+    if (
+      !isPacedTrainResponseWithPacedTrainId(timetableItem) ||
+      extractPacedTrainIdFromOccurrenceId(trainId) !== timetableItem.id
+    )
+      return undefined;
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { id, exceptions, paced, ...trainProps } = timetableItem;
```

Comment:
nit : we could also do that to avoid the "eslint disable"
```suggestion
    const { id: _id, exceptions: _exceptions, paced: _paced, ...trainProps } = timetableItem;
```

Reply:
Hum I prefer to keep the eslint disable instead of renaming all the unused props with an underscore :)

Reply:
Fine by me :)